assets/src/edit-story/components/mediaPicker/useMediaPicker.js

```diff
@@ -91,15 +91,22 @@ export default function useMediaPicker({
     });
 
     // When an image is selected, run a callback.
-    fileFrame.on('select', () => {
+    fileFrame?.once('select', () => {
       const mediaPickerEl = fileFrame.state().get('selection').first().toJSON();
       onSelect(mediaPickerEl);
     });
 
     if (onClose) {
-      fileFrame.on('close', onClose);
+      fileFrame?.once('close', onClose);
     }
 
+    fileFrame?.once('content:activate:browse', () => {
+      // Force-refresh media modal contents every time
+      // to avoid stale data.
+      fileFrame?.content?.get()?.collection?._requery(true);
```

Comment:
Don't we also need to refresh media library if you drag an image/video into the editor?

Reply:
Yes, but this code already handles this, because it refreshes when opening the modal.

1. Drag video into editor to upload
2. Open media modal
3. Media modal refreshes automatically, and will include the new video